static/src/javascripts/projects/commercial/modules/brazeBanner.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+
+import type {Banner} from 'common/modules/ui/bannerPicker';
+import config from 'lib/config';
+import reportError from 'lib/report-error';
+import {onConsentChange} from '@guardian/consent-management-platform';
+import {mountDynamic} from "@guardian/automat-modules";
+
+import {getUserFromApi} from '../../common/modules/identity/api';
+import {isDigitalSubscriber} from "../../common/modules/commercial/user-features";
+
+const brazeSwitch = config.get('switches.brazeSwitch');
+const apiKey = config.get('page.brazeApiKey');
+const brazeVendorId = '5ed8c49c4b8ce4571c7ad801';
+
+const getBrazeUuid = (): Promise<?string> =>
+    new Promise((resolve) => {
+        getUserFromApi(user => {
+            if (user && user.privateFields && user.privateFields.brazeUuid) {
+                resolve(user.privateFields.brazeUuid);
+            } else {
+                resolve();
+            }
+        })
+    });
+
+const hasRequiredConsents = (): Promise<boolean> =>
+    new Promise((resolve) => {
+        onConsentChange(({ tcfv2, ccpa }) => {
+            const consentGivenUnderCcpa = ccpa && !ccpa.doNotSell;
+            const consentGivenUnderTcfv2 = tcfv2 && tcfv2.vendorConsents[brazeVendorId];
+
+            resolve(!!(consentGivenUnderCcpa || consentGivenUnderTcfv2));
```

Comment:
I'd be tempted to ensure `consentGivenUnderCcpa` and `consentGivenUnderTcfv2` are already booleans when they reach here, rather than coercing the result (with `!!`) at this late stage.

Reply:
That sounds sensible. Because of the short circuiting just above I think this would imply moving the `!!` coersion up to those two lines (i.e. where we define `const consentGivenUnderCcpa` and `const consentGivenUnderTcfv2`).

Reply:
Addressed this in b164954. I rewrote this in a way I think makes it clearer (and doesn't require any coercing). It now looks similar to the code in the cmp docs.